sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -510,16 +484,11 @@ case class Logarithm(left: Expression, right: Expression)
   }
 
   override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
-    val logCode = if (left.isInstanceOf[EulerNumber]) {
+    if (left.isInstanceOf[EulerNumber]) {
       defineCodeGen(ctx, ev, (c1, c2) => s"java.lang.Math.log($c2)")
     } else {
       defineCodeGen(ctx, ev, (c1, c2) => s"java.lang.Math.log($c2) / java.lang.Math.log($c1)")
     }
-    logCode + s"""
-      if (Double.isNaN(${ev.primitive})) {
-        ${ev.isNull} = true;
-      }
-    """
```

Comment:
remove the test here since two NaN cannot be compared, only test with `isNaN`, since we've check NaN in `MathFunctionSuite`, I think it's just safe to delete this.